src/fsharp/FSComp.txt

```diff
@@ -17,6 +17,7 @@ undefinedNameTypeParameter,"The type parameter %s is not defined."
 undefinedNamePatternDiscriminator,"The pattern discriminator '%s' is not defined."
 replaceWithSuggestion,"Replace with '%s'"
 missingElseBranch,"The 'if' expression is missing an 'else' branch. The 'then' branch has type '%s'. Because 'if' is an expression, and not a statement, add an 'else' branch which returns a value of the same type."
+ifExpression,"The 'if' expression needs to return type '%s' in order to satisfy context type requirements. It currently returns type '%s'."
```

Comment:
mmm if you have to ask then it's probably not good. 
I wanted to tell that the type requirement comes from the context of the expression - not from the expression itself.